libs/storage/Tsavorite/cs/src/core/TsavoriteLog/TsavoriteLog.cs

```diff
@@ -872,28 +861,20 @@ private long AllocateBlock(int recordSize)
             while (true)
             {
                 var flushEvent = allocator.FlushEvent;
-                var logicalAddress = allocator.TryAllocate(recordSize);
+                var logicalAddress = allocator.TryAllocateRetryNow(recordSize);
                 if (logicalAddress > 0)
                     return logicalAddress;
-
-                if (logicalAddress == 0)
+                Debug.Assert(logicalAddress == 0);
```

Comment:
Add a message to the Assert e.g. "<0 (RETRY_NOW) should already have been handled"